Homework7: Anton Stychnevsky

Stychnevsky opened the pull request:
Homework # 7

Stychnevsky committed:
Homework # 7

Stychnevsky committed:
add requirements.txt

Stychnevsky committed:
fixed according PEP8

Stychnevsky committed:
Add some more tests and create separate function for any test

belovmd commented on HW7_task1_3.py:
```diff
@@ -0,0 +1,22 @@
+        print("Created %s." % listname)
+    else:
+        print("%s already has %d elements." % (listname, len(curr_list)))
+    finally:
```
redundant finally we can not use it here

belovmd commented on test_hw.py:
```diff
@@ -0,0 +1,123 @@
+
+
+class TestHomework7AddToListInDict(unittest.TestCase):
+    ex_dict = {'key1': [1, 2], 'key2': [4]}
```
It's not good to use mutable dict as shared var

belovmd commented on test_hw.py:
```diff
@@ -0,0 +1,123 @@
+
+    def test_character_with_count_1(self):
+        with patch('builtins.input', return_value="abcdefbbc"):
+            self.assertEqual(HWs2_5.count_characters()['a'], 1)
```
it seems we should assert whole dicts

belovmd commented on test_hw.py:
```diff
@@ -0,0 +1,123 @@
+            HWs2_5.generate_numbers('twenty')
+
+
+class TestRunner(unittest.TestCase):
```
we can also test w/o parameters

belovmd commented on test_hw.py:
```diff
@@ -0,0 +1,123 @@
+        self.assertEqual(runner('fizzbuzz')[0][4], 'Buzz')
+
+    def test_generate_numbers(self):
+        self.assertEqual(runner('generate_numbers')[0][4], 16)
```
please assert whole expected results

belovmd reviewed (commented):
please address these issues. Usually we don't assert part of result

Stychnevsky committed:
refactored and add testing runner()

1) Change tests asserted part of result
Now all test assert full function result
2) Refactor TestHomework7AddToListInDict: delete mutable dict as shared var
3) Create test of runner() without parameters. Use mock to hide inputs it tests.

Stychnevsky committed:
Change assertRaises Exception to ValueError and TypeError

belovmd commented on test_hw.py:
```diff
@@ -0,0 +1,158 @@
+            self.assertEqual(HWs2_5.count_characters(), {'a': 1})
+
+    def test_empty_input(self):
+        with patch('builtins.input', return_value=""):
```
👍

belovmd commented on test_hw.py:
```diff
@@ -0,0 +1,158 @@
+                         )
+
+
+class TestGenerateNumbers(unittest.TestCase):
```
we can also check 0

belovmd reviewed (commented):
9,5
